### DISCUSSION TITLE
Publish to Confluence with dynamically specified page title

### DISCUSSION THREAD

**Discussion by mazanise** [quarto-dev/quarto-cli#5906] [Q&A]
### Description

The publish command does not seem to work with r code in the YAML headers.
I want to publish to Confluence and specify the title with R code. At the moment there are random numbers appended at the title of the Confluence pages. According to documentation this is done to ensure name uniqueness. I want to make the title of the Confluence page to be the current date:
```
title: "`r format(Sys.time(), '%d %B, %Y')`"
```
 However after running the `publish confluence` command the resulting Confluence page has this name: 
```
"`r format(Sys.time(), '%d %B, %Y')`"
```

**Comment by cscheid** [comment]
Is this your only command that executes R code? If so, you need to explicitly add

```yaml
engine: knitr
```

to your frontmatter.

**Comment by mazanise** [comment]
Thank you for your fast reply. This is my full document:
```
---
title: "`r format(Sys.time(), '%d %B, %Y')`"
format: confluence-html
engine: knitr
---
```{r echo=FALSE, results = 'asis'}
tables <- readRDS("tables.RDS")
tables[[1]]
tables[[2]]
tables[[3]]
tables[[4]]
```
So essentially I am publishing a bunch of tables on Confluence. 
Say the file is called `report.qmd`, then I run `quarto publish confluence report.qmd`. 
I just added what you said but get the same result. Not sure if I am missing something.

**Reply by cscheid** [reply]
Thanks. You're probably not missing anything, then, and it's a quarto bug.

**Reply by cscheid** [reply]
@allenmanning I wonder if resolving this sort of code dynamically is doable in our current confluence pipeline.

**Reply by ghost** [reply]
Thanks for reporting.  I've moved this [to a bug](https://github.com/quarto-dev/quarto-cli/issues/5916).

**Comment by ghost** [comment]
Moved to https://github.com/quarto-dev/quarto-cli/issues/5916
